src/CHANGES.txt

```diff
@@ -6,6 +6,9 @@
 
 RELEASE  VERSION/DATE TO BE FILLED IN LATER
 
+  From Daniel Moody:
+    - Add no_progress (-Q) option as a set-able option
```

Comment:
isn't -Q for silent option?

Reply:
It is, but during the test it's not being quiet - probably for the reson discussed above (you don't see the option setting until you're far enough in to read it from the sconscript file).  Same class of problems I'm having with the various broken option-reading things: where we have to parse the command line first because if might provide information that could impact the operation of an SConscrtip, but then the SConscript can add command-line options with `AddOption` calls - by which time some information that belonged to such options has already been snarfed into other places.  Very unfortunate chicken-and-egg...

Reply:
I'm thinking re-architecting the tools as a class which has a method to add args would be simpler than the reparsing we do? of course you'd have to note in help which tool it came from... still I think doable?